app/entities/user.go

```diff
@@ -16,9 +15,83 @@ type User struct {
 	CreatedAt string `json:"created_at"`
 }
 
-// TODO: sentinel errors
-
-// Validate will validate User's signup data
 func (u User) Validate() error {
+	if err := checkName(u.FirstName); err != nil {
+		return err
+	}
+	if u.LastName != "" {
+		if err := checkName(u.LastName); err != nil {
+			return err
+		}
+	}
+
+	if err := checkMail(u.Email); err != nil {
+		return err
+	}
+
+	if err := checkPass(u.Password); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkName(name string) error {
+	const (
+		minNameLen int = 2
+		maxNameLen int = 256
+	)
+	// Checking the length
+	nameLen := len(name)
+	switch {
+	case nameLen < minNameLen:
+		return errors.New("name is too short")
+	case nameLen > maxNameLen:
+		return errors.New("name is too long")
+	}
+	// Checking for forbidden symbols
+	if valid, _ := regexp.MatchString(`^[a-zA-Zа-яА-ЯҐІЇЄґіїє'\s]{2,256}$`, name); !valid {
```

Comment:
we don't need min and max length check if we have `{2,256}` in regEx mask

Reply:
Looks good but `Hose-Ingasio` and `Jorge Jr` would not pass this check

Reply:
In case we will leave min max check `^[\p{L}&\s-\\'’.]+$` or `^[\p{L}&\s-\\'’.]{2,256}$`
Just for your interest `\p{L}` - matches any kind of letter from any language